tempodb/tempodb.go

```diff
@@ -256,7 +258,22 @@ func (rw *readerWriter) Find(ctx context.Context, tenantID string, id common.ID,
 		return foundObject, nil
 	})
 
-	return foundBytes, metrics, err
+	if len(partialTraces) == 0 {
+		return nil, metrics, err
+	}
+
+	// merge all partial trace bytes into partialTraces[0]
+	for i := range partialTraces {
+		if i == 0 {
+			continue
+		}
+		partialTraces[0], err = tempo_util.CombineTraces(partialTraces[0], partialTraces[i])
```

Comment:
This will unmarshal/marshal the trace repeatedly for every partial.  It would be more efficient use CombineTraceProtos.  Probably not significant in practice (low likelihood).

Reply:
Yeah, I agree with this.  Let's have `store.Find` return a slice of all objects it finds.  Then the querier can marshal each to proto once and combine.

Reply:
Created a variadic version of CombineTraces which I think is what we need here:  https://github.com/mdisibio/tempo/blob/combinetracesv/pkg/util/trace.go#L17    This preserves all the error handling/logic of the current function, rather that doing it here in the Find method.

Reply:
I'm going to go ahead with the current function signature in this PR to avoid including too many changes, but I agree that we need this @mdisibio. I'll work on it in a follow up PR.